python/lsst/meas/algorithms/testUtils.py

```diff
@@ -178,3 +184,122 @@ def makeDefectList():
                   ]
 
     return defectList
+
+
+class MockRefcatDataId:
+    """Mock reference catalog dataId.
+
+    The reference catalog dataId is only used to retrieve a region property.
+
+    Parameters
+    ----------
+    pixelization : `lsst.sphgeom.Pixelization`
+        Pixelization object.
+    index : `int`
+        Pixel index number.
+    """
+    def __init__(self, pixelization, index):
+        self._region = pixelization.pixel(index)
+
+    @property
+    def region(self):
+        return self._region
+
+
+class MockRefcatDeferredDatasetHandle:
+    """Mock reference catalog dataset handle.
+
+    The mock handle needs a get() and a name for logging.
+
+    Parameters
+    ----------
+    catalog : `lsst.afw.table.SourceCatalog`
+        Reference catalog.
+    name : `str`
+        Name of reference catalog.
+    """
+    def __init__(self, catalog, name):
+        self._catalog = catalog
+        self._name = name
+
+    def get(self):
+        return self._catalog
+
+    class MockRef:
+        def __init__(self, name):
+            self._name = name
+
+        class MockDatasetType:
+            def __init__(self, name):
+                self._name = name
+
+            @property
+            def name(self):
+                return self._name
+
+        @property
+        def datasetType(self):
+            return self.MockDatasetType(self._name)
+
+    @property
+    def ref(self):
+        return self.MockRef(self._name)
+
+
+class MockReferenceObjectLoaderFromFiles(ReferenceObjectLoader):
+    """A simple mock of ReferenceObjectLoader to use a set of files.
```

Comment:
Oh, let'ss expand this docstring to something like: "A ReferenceObjectLoader that doesn't know about regions on the sky or the butler, and takes the list of files on disk to mock dataIds."

Reply:
```
    This mock ReferenceObjectLoader uses a set of files on disk to create
    mock dataIds and data reference handles that can be accessed
    without a butler. The files must be afw catalog files in the reference
    catalog format, sharded with HTM pixelization.
```